docs/modules/ROOT/pages/dapp.adoc

```diff
@@ -0,0 +1,337 @@
+----
+$ npm install @openzeppelin/solidity-loader json-loader
+----
+
+Next up, we configure Solidity Hot Loader and set it to enabled. To stop the Solidity Hot Loader automagically updating our contracts, set disabled to `true`.
+
+In the `client` directory create a `config` directory. In the `config` directory create a `webpack.js` file for the Solidity Hot Loader webpack rule with the following code, notice that we are starting with the Solidity Hot Loader enabled (`disabled: false`):
+
+[source,javascript]
+----
+const solidityLoaderOptions = {
+  network: 'development',
+  // you can stop loader from automatic compile/push/upgrade
+  // action by setting disabled flag to true, but it will still
+  // serve .json files from file system
+  disabled: false,
+};
+
+module.exports = {
+  solidityLoader: {
+    test: /\.sol$/,
+    use: [
+      { loader: 'json-loader' },
+      {
+        loader: '@openzeppelin/solidity-loader',
+        options: solidityLoaderOptions,
+      },
+    ],
+  },
+  solidityLoaderOptions,
+};
+----
+
+We now add in the Solidity Hot Loader to webpack.  In the `client` directory create a `config-overrides.js` file with the following code:
+
+[source,javascript]
+----
+/* config-overrides.js */
+const { solidityLoader } = require('./config/webpack');
+
+module.exports = function override(config, env) {
+  //do stuff with the webpack config...
+
+  // allow importing from outside of app/src folder, ModuleScopePlugin prevents this.
+  const scope = config.resolve.plugins.findIndex(o => o.constructor.name === 'ModuleScopePlugin');
+  if (scope > -1) {
+    config.resolve.plugins.splice(scope, 1);
+  }
+
+  // add solidity loader support
+  // have to insert before last loader, because CRA user 'file-loader' as default one
+  config.module.rules.splice(config.module.rules - 2, 0, solidityLoader);
+
+  return config;
+};
+----
+
+In this guide we use https://github.com/timarney/react-app-rewired[`react-app-rewired`] to rewire `create-react-app` webpack configs to add in the Solidity Hot Loader without ejecting from `create-react-app`.
+
+In the `client` directory, install `react-app-rewired`
+
+[source,console]
+----
+$ npm i react-app-rewired
+----
+
+To use `react-app-rewired`, we now need to change the `package.json` scripts.  In `package.json` in the `client` directory, update the scripts to use `react-app-rewired` so that they looks as follows:
+
+[source,javascript]
+----
+  "scripts": {
+    "start": "react-app-rewired start",
+    "build": "react-app-rewired build",
+    "test": "react-app-rewired test",
+    "eject": "react-scripts eject"
+  },
+----
+
+== Add the dapp code
+
+We will use https://github.com/OpenZeppelin/openzeppelin-network.js[OpenZeppelin Network.js] to access the Web3 provider. *OpenZeppelin Network.js* is an easy to use and reliable library that provides one line access to the web3 API. It can be used in both React and in (vanilla) JavaScript. It also supports the Gas Station Network.
```

Comment:
```suggestion
We will use xref:network-js::index.adoc[[OpenZeppelin Network JS*] to access the Web3 provider. This is an easy to use and reliable library that provides one line access to the web3 API. It can be used in both React and (vanilla) JavaScript, and it also supports the Gas Station Network.
```